systemtest/src/test/java/io/strimzi/systemtest/cruisecontrol/CruiseControlST.java

```diff
@@ -639,6 +642,141 @@ void testKafkaRebalanceAutoApprovalMechanism() {
+                        .withAutoRebalance(
+                            new KafkaAutoRebalanceConfigurationBuilder()
+                                .withMode(KafkaRebalanceMode.ADD_BROKERS)
+                                .withNewTemplate(scaleUpKafkaRebalanceTemplateName)
+                                .build(),
+                            new KafkaAutoRebalanceConfigurationBuilder()
+                                .withMode(KafkaRebalanceMode.REMOVE_BROKERS)
+                                .withNewTemplate(scaleDownKafkaRebalanceTemplateName)
+                                .build())
+                    .endCruiseControl()
+                .endSpec()
+            .build(),
+            KafkaTopicTemplates.topic(testStorage.getNamespaceName(), testStorage.getTopicName(), testStorage.getClusterName(), 12, 3).build(),
+            ScraperTemplates.scraperPod(testStorage.getNamespaceName(), testStorage.getScraperName()).build()
+        );
+
+        final String scraperPodName = kubeClient().listPodsByPrefixInName(testStorage.getNamespaceName(), testStorage.getScraperName()).get(0).getMetadata().getName();
+
+        // check that Kafka has `Idle` in auto-rebalance status
+        KafkaUtils.waitUntilKafkaHasStateAutoRebalance(testStorage.getNamespaceName(), testStorage.getClusterName(), KafkaAutoRebalanceState.Idle);
+
+        final int scaleTo = 5;
+        Map<String, String> ccPod = DeploymentUtils.depSnapshot(testStorage.getNamespaceName(), CruiseControlResources.componentName(testStorage.getClusterName()));
+
+        LOGGER.info("Scaling Kafka up to {}", scaleTo);
+        if (Environment.isKafkaNodePoolsEnabled()) {
+            KafkaNodePoolResource.replaceKafkaNodePoolResourceInSpecificNamespace(testStorage.getNamespaceName(), testStorage.getBrokerPoolName(), knp ->
+                knp.getSpec().setReplicas(scaleTo));
+        } else {
+            KafkaResource.replaceKafkaResourceInSpecificNamespace(testStorage.getNamespaceName(), testStorage.getClusterName(), kafka -> kafka.getSpec().getKafka().setReplicas(scaleTo));
+        }
+
+        // 3 brokers + 3 controllers/zk
+        final int kafkaClusterPodIndex = initialReplicas + initialReplicas;
+
+        KafkaUtils.waitUntilKafkaHasExpectedModeAndBrokersAutoRebalance(testStorage.getNamespaceName(), testStorage.getClusterName(),
+            KafkaRebalanceMode.ADD_BROKERS,
+            // brokers with [6, 7]
+            Arrays.asList(kafkaClusterPodIndex, kafkaClusterPodIndex + 1));
+
+        // check that Kafka has `RebalanceOnScaleUp` in auto-rebalance status
+        KafkaUtils.waitUntilKafkaHasStateAutoRebalance(testStorage.getNamespaceName(), testStorage.getClusterName(), KafkaAutoRebalanceState.RebalanceOnScaleUp);
+
+        // and then afterward we again move to Idle state after all is done
+        KafkaUtils.waitUntilKafkaHasStateAutoRebalance(testStorage.getNamespaceName(), testStorage.getClusterName(), KafkaAutoRebalanceState.Idle);
+
```

Comment:
I would also add the check that the corresponding KafkaRebalance custom resource for CC was deleted.